scipy/differentiate/tests/test_differentiate.py

```diff
@@ -547,103 +552,113 @@ def df5(x):
     f5.mn = 3, 3  # type: ignore[attr-defined]
     f5.ref = df5  # type: ignore[attr-defined]
 
-    rosen = optimize.rosen
+    def rosen(x, _): optimize.rosen(x)
     rosen.mn = 5, 1  # type: ignore[attr-defined]
     rosen.ref = optimize.rosen_der  # type: ignore[attr-defined]
 
+    @pytest.mark.filterwarnings("ignore::UserWarning")
     @pytest.mark.parametrize('size', [(), (6,), (2, 3)])
     @pytest.mark.parametrize('func', [f1, f2, f3, f4, f5, rosen])
-    def test_examples(self, size, func):
+    def test_examples(self, size, func, xp):
+        one = xp.asarray(1.)
         rng = np.random.default_rng(458912319542)
         m, n = func.mn
         x = rng.random(size=(m,) + size)
-        res = jacobian(func, x)
-        ref = func.ref(x)
-        np.testing.assert_allclose(res.df, ref, atol=1e-10)
+        res = jacobian(lambda x: func(x , xp), xp.asarray(x, dtype=one.dtype))
+        ref = xp.asarray(np.asarray(func.ref(x)), dtype=one.dtype)
```

Comment:
What does the nested NumPy based coercion do inside the array API-based conversion?

Locally I seem to get the same testing behavior if I simplify, but maybe I'm missing something:

```diff
--- a/scipy/differentiate/tests/test_differentiate.py
+++ b/scipy/differentiate/tests/test_differentiate.py
@@ -565,7 +565,7 @@ class TestJacobian(JacobianHessianTest):
         m, n = func.mn
         x = rng.random(size=(m,) + size)
         res = jacobian(lambda x: func(x , xp), xp.asarray(x, dtype=one.dtype))
-        ref = xp.asarray(np.asarray(func.ref(x)), dtype=one.dtype)
+        ref = xp.asarray(func.ref(x), dtype=one.dtype)
         xp_assert_close(res.df, ref, atol=1e-10)
```

Reply:
I got the same set of failures before and after the change with `torch` as well.

Reply:
might be worth a comment if it is specific to one of the skipped backends or something?

Reply:
Added.